build_pros.cmake

```diff
@@ -0,0 +1,241 @@
+# ======== Policies ========
+
+if(POLICY CMP0075) # Include file check macros honor CMAKE_REQUIRED_LIBRARIES, CMake >= 3.12
+    cmake_policy(SET CMP0075 NEW)
+endif()
+
+if(POLICY CMP0077) # Cache variables override since 3.12
+    cmake_policy(SET CMP0077 NEW)
+endif()
+
+# ======== Includes ========
+
+include(CheckSymbolExists)
+include(CheckFunctionExists)
+include(CheckIncludeFile)
+include(CMakePushCheckState)
+include(CheckCSourceCompiles)
+include(TestBigEndian)
+
+# If platform is Emscripten
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Emscripten")
+    set(EMSCRIPTEN 1)
+endif()
+
+
+# ======== Optimizations and warnings ========
+
+# Strip garbage
+if(APPLE)
+    string(REGEX REPLACE "-O3" ""
+        CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O2")
+    set(LINK_FLAGS_RELEASE  "${LINK_FLAGS_RELEASE} -dead_strip")
+
+    # Unify visibility to meet llvm's default.
+    include(CheckCCompilerFlag)
+    check_c_compiler_flag("-fvisibility-inlines-hidden" SUPPORTS_FVISIBILITY_INLINES_HIDDEN_FLAG)
```

Comment:
1. Check this with `-Werror` in clfags. I have this in my log:
    `cc1: warning: command line option "-fvisibility-inlines-hidden" is valid for C++/ObjC++ but not for C`
2. Do we _truly_ need this?
3. If we do, move it among the other visibility checks (and please change that 2km long
   `SUPPORTS_FVISIBILITY_INLINES_HIDDEN_FLAG` to `HAVE_VISIBILITY_INLINES_HIDDEN`
   or something.)

Reply:
>  cc1: warning: command line option "-fvisibility-inlines-hidden" is valid for C++/ObjC++ but not for C

that was a code piece I used for C++-targeted project sides, forgot that is C++only, gonna remove this.

Reply:
Removed this.